src/analyzer.ts

```diff
@@ -107,37 +173,34 @@ const exportWildCards = (file: SourceFile) =>
       type: AnalysisResultTypeEnum.DEFINITELY_USED
     }));
 
-const emitDefinitelyUsed = (file: SourceFile, onResult: OnResultType) => {
-  [
-    ...importWildCards(file),
+const getDefinitelyUsed = (file: SourceFile): IAnalysedResult[] => ([
+    ...importsForSideEffects(file),
     ...exportWildCards(file),
-  ].forEach(onResult);
-};
+  ]);
 
-const emitPotentiallyUnused = (file: SourceFile, onResult: OnResultType) => {
+export const getPotentiallyUnused = (file: SourceFile): IAnalysedResult => {
   const exported = getExported(file);
 
-  const referenced2D = file
+  const idsInFile = file.getDescendantsOfKind(ts.SyntaxKind.Identifier);
+  const referenceCounts = countBy(x => x)((idsInFile || []).map(node => node.getText()));
+  const referencedInFile = Object.entries(referenceCounts).flatMap(([name, count]) => count > 1 ? [name] : []);
+
+  const referenced = file
     .getReferencingNodesInOtherSourceFiles()
-    .map((node: SourceFileReferencingNodes) => {
-      const handler =
-        nodeHandlers[node.getKind().toString()] ||
-        function noop() {
-          return [] as string[];
-        };
-
-      return handler(node);
+    .flatMap((node: SourceFileReferencingNodes) => {
+      const kind = node.getKind().toString();
+      return nodeHandlers?.[kind]?.(node) ?? [];
     });
 
-  const referenced = ([] as string[]).concat(...referenced2D);
-
-  const unused = referenced.includes("*") ? [] : exported.filter(exp => !referenced.includes(exp.name));
+  const unused = referenced.includes("*") ? [] :
+    exported.filter(exp => !referenced.includes(exp.name))
+    .map(exp => ({...exp, usedInModule: referencedInFile.includes(exp.name)}))
 
-  onResult({
-    file: realpathSync(file.getFilePath()),
```

Comment:
I couldn't figure out why the `realpathSync` call was needed here and it prevented the tests in `analyzer.test.ts`, which use in-memory files, from running. Happy to revert & work out a different approach for the tests if this is needed.

Reply:
I think the reason for using it is documented in #66 

I think you've got a couple of options here.

You can either keep the scope of the unit tests to testing the functionality of individual methods, and leave the black box testing for integration tests.

Or, you can mock the `realpathSync` to be `path => path` in the test.

My preference is for the former, but I'm happy for you to make the call.

Reply:
I consolidated the calls to `realPathSync` into `analyze`, which is not hit directly by the unit test. I think this should meet both the need to unit test without writing to disk & handle symlinks.

It would be helpful if this symlink case were tested by the integration test!